File: Problems/3174. Clear Digits/Solution.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    string clearDigits(string s) {
        stack<char> st;
        
        for (char c : s) {
            if (isdigit(c) && !st.empty()) {
                st.pop();
            }
            else {
                st.push(c);
            }
        }
        
        string result;
        while (!st.empty()) {
            result += st.top();
            st.pop();
        }
        reverse(result.begin(), result.end());
        return result;
    }
};

File: Problems/3174. Clear Digits/Solution.rs
impl Solution {
    pub fn clear_digits(s: String) -> String {
        let mut stack: Vec<char> = Vec::new();

        for c in s.chars() {
            if c.is_ascii_digit() && !stack.is_empty() {
                stack.pop();
            }
            else {
                stack.push(c);
            }
        }

        return stack.iter().collect();
    }
}